### PR TITLE
fix doxygen warnings in cudf/io/ avro, csv, json, orc, parquet header files

### PR DESCRIPTION
**Pull request by karthikeyann**
Fixes parts of https://github.com/rapidsai/cudf/issues/9373
added missing documentation to fix doxygen warnings in multiple files
- cpp/include/cudf/io/avro.hpp
- cpp/include/cudf/io/csv.hpp
- cpp/include/cudf/io/json.hpp
- cpp/include/cudf/io/orc.hpp
- cpp/include/cudf/io/orc_metadata.hpp
- cpp/include/cudf/io/parquet.hpp

fixes 194 warnings

### CODE REVIEW

File: cpp/include/cudf/io/avro.hpp [hunk 1]
```diff
@@ -73,21 +70,29 @@ class avro_reader_options {
 
   /**
    * @brief Returns source info.
+   *
+   * @return Source info
```
Comment (PointKernel) [inline cpp/include/cudf/io/avro.hpp:74]: `@return` basically repeats what's been explained in `@brief`. This kind of warning seems an "overkill" for simple getters like here.

File: cpp/include/cudf/io/parquet.hpp [hunk 2]
```diff
@@ -95,12 +92,16 @@ class parquet_reader_options {
...
+   *
+   * @return Source info
    */
   [[nodiscard]] source_info const& get_source() const { return _source; }
 
   /**
    * @brief Returns true/false depending on whether strings should be converted to categories or
    * not.
+   *
+   * @return True/false depending on whether strings should be converted to categories or not.
```
Comment (PointKernel) [inline cpp/include/cudf/io/parquet.hpp:104]:
```suggestion
   * @return `true` if strings should be converted to categories.
```
Would this be easier to follow? The same for other boolean returns.

**Review by PointKernel — COMMENTED** [review]
The burn down started before the creation of this PR. Should we target the work for the next release?

File: cpp/include/cudf/io/avro.hpp (hunk 1 above)
Comment (karthikeyann) [reply cpp/include/cudf/io/avro.hpp:74]: There is no way to differentiate between simple getters and legitimate `@return` missing.
at the moment, doxygen has no way to differentiate or simplify this. 

Aliases is one solution to reduce verbose typing. but not recommended so far in reviews https://github.com/rapidsai/cudf/pull/10861#pullrequestreview-975990743

File: cpp/include/cudf/io/parquet.hpp (hunk 2 above)
Comment (karthikeyann) [reply cpp/include/cudf/io/parquet.hpp:104]: will update this.

**Comment by karthikeyann** [comment]
> The burn down started before the creation of this PR. Should we target the work for the next release?

This PR only has changes only in comments. Since entire documentation cleanup will take more PRs, we can move target to next release.

**Review by PointKernel — APPROVED** [review]

**Comment by jlowe** [comment]
I removed the Java review request, as there aren't any Java files modified.  I assume this was accidentally triggered by the rebase to 22.08.

**Comment by jlowe** [comment]
Removed ops and cmake review for similar reasons.

**Comment by karthikeyann** [comment]
rerun tests

File: conda/recipes/libcudf/meta.yaml [hunk 3]
```diff
@@ -183,7 +183,6 @@ outputs:
         - test -f $PREFIX/include/cudf/lists/gather.hpp
         - test -f $PREFIX/include/cudf/lists/list_view.hpp
         - test -f $PREFIX/include/cudf/lists/lists_column_view.hpp
-        - test -f $PREFIX/include/cudf/lists/list_view.hpp
```
Comment (vyasr) [inline conda/recipes/libcudf/meta.yaml:186]: Why is this removed?

**Review by vyasr — APPROVED** [review]
A lot of these docstrings would be nice if they could be `copydoc`ed from a single source of truth, but I'll leave it up to you if you think that's a good idea or not since I requested something similar on a couple of your other PRs.

File: conda/recipes/libcudf/meta.yaml (hunk 3 above)
Comment (ttnghia) [reply conda/recipes/libcudf/meta.yaml:186]: This is duplicate of line 184 above.

**Review by ttnghia — APPROVED** [review]

**Comment by karthikeyann** [comment]
@gpucibot merge

**Comment by karthikeyann** [comment]
@gpucibot merge
